mindmeld/components/nlp.py

```diff
@@ -1396,6 +1399,38 @@ def process_query(self, query, allowed_nlp_classes=None, dynamic_resource=None,
             query, dynamic_resource=dynamic_resource, verbose=verbose
         )
 
+        # This code block implements allowed entities described here:
+        # https://github.com/cisco/mindmeld/pull/280
+        if allowed_nlp_classes and all(entity == () for entity in entities):
+            dynamic_gazetteer = dynamic_resource.get(GAZETTEER_RSC, {}) if dynamic_resource else {}
+            entities = []
+            for n_best_query in query:
+                n_best_entities = []
+                for entity in allowed_nlp_classes:
+                    # check if entity is in the gazetteers
+                    if n_best_query.text in {
```

Comment:
It may be the case that an entity narrowly misses a stored value in the gazetteer. This could be the result of a typo or ASR mistranscription. For example, "chcking". While this can be addressed by adding misspellings in the gazetteer, the can become tedious given the space of possible misspellings and number of entities. Instead, we can have this robustness in-built using edit distance calculations from NLTK. An EDIT_DISTANCE_THRESHOLD of 1 can be set internally. On the flip side, there may be some edge cases, especially with smaller words, where a word is incorrectly considered a gazetteer due to a small edit distance. 

```
>> import nltk
>> nltk.edit_distance("chckings", "checkings")
>> 1
```

Reply:
I agree fuzzy matching will improve the functionality of this application. However, I would like to move the scope of this to another PR and once we have product validation of it. It doesnt seem trivial both performance-wise and hyperparameter-wise when we choose the correct edit distance.